### PR TITLE
feat(Datatable): add Table component

### PR DESCRIPTION
**Pull request by ajkl2533**

### CODE REVIEW

File: src/components/DatatableV2/Table/Table.tsx [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+const Table = <D extends Record<string, unknown>>({
+  columns,
+  data,
+  dataPrimaryKey,
+  dataSize,
+  isDataLoading,
+  rowActions,
+  NoMatchingDataComponent,
+  NoDataComponent,
```
Comment (youssefssc) [inline src/components/DatatableV2/Table/Table.tsx:63]: Both the default components here are taken from `defaultTableConfig` but they are optional props. Perhaps better to move the default values here?

File: src/components/DatatableV2/Table/Table.reducer.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+}: {
+  collectFetchParams: (
+    pageIndex: number,
+    pageSize: number,
+    sortBy: SortingRule<D>[],
+  ) => void;
+  scrollToTableTop: () => void;
+}): StateReducerFn<D> {
+  return (newState, action) => {
+    // console.log('New State', newState);
```
Comment (PrincAm) [inline src/components/DatatableV2/Table/Table.reducer.ts:19]: commented code

**Review by youssefssc — APPROVED** [review]
Nice!

**Comment by ajkl2533** [comment]
@youssefssc nice catch will fix that

**Pull request merged**